ethcontract-generate/src/contract/common.rs

```diff
@@ -174,6 +175,20 @@ fn expand_address(address: Address) -> TokenStream {
     }
 }
 
+/// Expands a transaction hash into a literal representation that can be used
+/// with quasi-quoting for code generation.
+fn expand_transaction_hash(hash: Option<TransactionHash>) -> TokenStream {
+    let hash = match hash {
+        Some(hash) => hash,
+        None => return quote! { None },
+    };
+
+    let bytes = hash.as_bytes().iter().copied().map(Literal::u8_unsuffixed);
+    quote! {
+        self::ethcontract::TransactionHash([#( #bytes ),*])
```

Comment:
This might require re-exporting `H256 as TransactionHash` in `ethcontract/src/lib.rs` as well.